plugins/services/src/js/components/forms/EnvironmentFormSection.js

```diff
@@ -0,0 +1,213 @@
+import React, {Component} from 'react';
+
+import FieldError from '../../../../../../src/js/components/form/FieldError';
+import FieldInput from '../../../../../../src/js/components/form/FieldInput';
+import FieldLabel from '../../../../../../src/js/components/form/FieldLabel';
+import FormGroup from '../../../../../../src/js/components/form/FormGroup';
+import ReducerUtil from '../../../../../../src/js/utils/ReducerUtil';
+import {FormReducer as env} from '../../reducers/serviceForm/EnvironmentVariables';
+import {FormReducer as labels} from '../../reducers/serviceForm/Labels';
+
+const METHODS_TO_BIND = [
+  'handleOnRemoveItem',
+  'handleOnAddItem'
+];
+
+class EnvironmentFormSection extends Component {
+  constructor() {
+    super(...arguments);
+
+    let reducers = ReducerUtil.combineReducers(EnvironmentFormSection.configReducers);
+
+    this.state = {reducers};
+
+    METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  handleOnRemoveItem(event) {
+    let {path, value} = event.target.dataset;
+
+    value = parseInt(value, 10);
+    this.props.onRemoveItem({value, path});
+  }
+
+  handleOnAddItem(event) {
+    let {path, value} = event.target.dataset;
+
+    value = parseInt(value, 10);
+    this.props.onAddItem({value, path});
+  }
+
+  getEnvironmentLines(data) {
+    const {errors} = this.props;
+
+    return data.map((env, key) => {
+      let keyLabel = null;
+      let valueLabel = null;
+      if (key === 0) {
+        keyLabel = <FieldLabel>Key</FieldLabel>;
+        valueLabel = <FieldLabel>Value</FieldLabel>;
+      }
+      return (
+        <div key={key} className="flex row">
+          <FormGroup
+            className="column-3"
+            required={false}
+            showError={Boolean(errors.env[key])}>
+            {keyLabel}
+            <FieldInput
+              name={`env.${key}.key`}
+              type="text"
+              value={env.key}/>
+            <FieldError>{errors.env[key]}</FieldError>
+            <span className="emphasis form-colon">:</span>
+          </FormGroup>
+          <FormGroup
+            className="column-3"
+            required={false}
+            showError={Boolean(errors.env[key])}>
+            {valueLabel}
+            <FieldInput
+              name={`env.${key}.value`}
+              type="text"
+              value={env.value}/>
+            <FieldError>{errors.env[key]}</FieldError>
+          </FormGroup>
+          <div className="form-group flex flex-item-align-end column-4">
+            <a className="column-3 button button-primary-link button-flush"
```

Comment:
Let's remove the `column-3` class here. `button-flush` isn't removing the padding because we're on an outdated version of CNVS, but once that's up to date, the button should be aligned nicely :)